platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/MapPropertyIntegrationTest.groovy

```diff
@@ -195,6 +195,48 @@ class MapPropertyIntegrationTest extends AbstractIntegrationSpec {
         failure.assertHasCause("The value for task ':thing' property 'prop' is final and cannot be changed any further.")
     }
 
+    def "UPGRADED task @Input property is NOT implicitly finalized when task starts execution"() {
```

Comment:
❓ Shall the names of these new tests be phrased in a way to indicate that this is a temporary behavior? How about:

```suggestion
    def "UPGRADED task @Input property is LENIENTLY finalized when task starts execution UNTIL NEXT MAJOR"() {
```

Reply:
We can also use that, changed as suggested